frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -185,6 +215,92 @@ const MenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
 function MainMenu(props: Props): ReactElement {
   const isServerDisconnected = !props.isServerConnected
 
+  const onClickDeployApp = useCallback((): void => {
+    const {
+      deployParams,
+      showDeployError,
+      closeDialog,
+      isDeployErrorModalOpen,
+    } = props
+
+    if (!deployParams) {
+      const dialog = NoRepositoryDetected()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
```

Comment:
Not a big deal, but you can probably just put the return all on one line.

`return showDeplyError(...)` for each one.

The result is exactly the same, but the code is a little tighter. I'm assuming the newlines are linting errors/formatter settings.